samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -324,4 +398,45 @@ public void subscribeToProcessorChange(IZkChildListener listener) {
     zkClient.subscribeChildChanges(keyBuilder.getProcessorsPath(), listener);
     metrics.subscriptions.inc();
   }
+
+  /**
+   * Represents zookeeper processor node.
+   */
+  private static class ProcessorNode {
```

Comment:
Discussed offline. 
`ProcessorData` represents actual data(processorId, hostname) stored in zkNode. 
`ProcessorNode` holds auxiliary details about zkNode(currently processorpath, in future : metadata, parentNode, childNodes, ACL) along with `ProcessorData` .